modules/controller/dc-controller.js

```diff
@@ -365,6 +366,55 @@ class DCController {
             dv_node_id,
         );
     }
+
+    /**
+     * Query local data
+     * @param query Query
+     * @returns {Promise<*>}
+     */
+    async queryLocal(req, res) {
+        this.logger.api('POST: Query local request received.');
+        if (!req.body) {
+            res.status(400);
+            res.send({
+                message: 'Body is missing',
+            });
+            return;
+        }
+
+        const { query } = req.body;
+
+        this.logger.info(`Local query handling triggered with ${JSON.stringify(query)}.`);
+        const validationError = ObjectValidator.validateSearchQueryObject(query);
+        if (validationError) {
+            throw validationError;
+        }
+
+        const { path } = query[0];
+        const valuesArray = Utilities.arrayze(query[0].value);
+
+        const result = await this.graphStorage.findLocalQuery({
+            idType: path,
+            identifierKey: valuesArray,
+        });
+        const response = this.importService.packLocalQueryData(result);
+        for (let i = 0; i < response.length; i += 1) {
+            response[i].offers = [];
+            for (let j = 0; j < response[i].datasets.length; j += 1) {
+                // eslint-disable-next-line no-await-in-loop
+                const offer = await Models.offers.findOne({
```

Comment:
Did you try to do local query on dh side? Offer id should be fetched from bids table since dh doesn't have offer id in offers table.

Reply:
Also, we should find offer for dataset id that doesn't have failed status. Since we can have multiple offers for the same dataset_id

Reply:
Good point

Reply:
response: {
...
datasets: [
{
dataset_id = '',
offer_id=''
}
]
...
}